RecoParticleFlow/PFClusterProducer/plugins/ECAL2DPositionCalcWithDepthCorr.cc

```diff
@@ -161,7 +160,7 @@ calculateAndSetPositionActual(reco::PFCluster& cluster) const {
       if( rh_energy > 0.0 ) 
 	weight = rh_energy/cluster.energy();
 
-      const CaloCellGeometry* cell = ecal_geom->getGeometry(refhit->detId());
+      const CaloCellGeometry* cell = ecal_geom->getGeometry(refhit->detId()).get();
```

Comment:
If the `getGeometry` call returns a `std::shared_ptr` the the object pointed to by that `std::shared_ptr` can be deleted at the end of this line leaving `cell` pointing to invalid memory. `cell` needs to become a `std::shared_ptr` to guarantee the lifetime of the cell object.